### PR TITLE
fix(gatsby-core-utils): fix fetchRemoteFile when called in main process after being called in worker

### PR DESCRIPTION
**Pull request by pieh**
## Description

There are some issues with downloads, especially if PQR worker did download a file, and later main process attempts to download it (it will receive `undefined` instead of filepath)

In particular after file was downloaded by a worker (and therefore there is `cacheEntry` for given url already) and `fetchRemoteFile` is called in MAIN process for same url https://github.com/gatsbyjs/gatsby/blob/a67d339ff87ad0db5a519594ad1ecc8afd1c5be7/packages/gatsby-core-utils/src/fetch-remote-file.ts#L234-L246 will return undefined because of https://github.com/gatsbyjs/gatsby/blob/a67d339ff87ad0db5a519594ad1ecc8afd1c5be7/packages/gatsby-core-utils/src/fetch-remote-file.ts#L102-L111

I see 2 options here:
 - (one used right now in PR) use cache locks for all processes (main and workers)
 - just wrap "second" `pollUntilComplete` (linked above) with `if (IS_WORKER)`

### CODE REVIEW

File: packages/gatsby-core-utils/src/fetch-remote-file.ts [hunk 1]
```diff
@@ -236,14 +229,12 @@ async function fetchFile({
...
-    }
+    await cache.set(cacheIdForWorkers(url), {
+      status: `complete`,
+      result: filename,
+      workerId: WORKER_ID,
+      buildId: BUILD_ID,
+    })
 
     return filename
   } catch (err) {
```
Comment (vladar) [inline packages/gatsby-core-utils/src/fetch-remote-file.ts:240]: Should we also remove `if (IS_WORKER)` from the catch block?

Comment (pieh) [reply packages/gatsby-core-utils/src/fetch-remote-file.ts:240]: Ah, yep

Comment (pieh) [reply packages/gatsby-core-utils/src/fetch-remote-file.ts:240]: Adjusted in https://github.com/gatsbyjs/gatsby/pull/33932/commits/d98f51095907d44e2622a3c1851c5911ee0bf645

**Review by vladar — APPROVED** [review]
Makes sense to me 👍

**Comment by axe312ger** [comment]
Thanks!

**Comment by LekoArts** [comment]
```
Successfully published:
 - babel-plugin-remove-graphql-queries@4.1.2
 - babel-preset-gatsby@2.1.2
 - gatsby-cli@4.1.3
 - gatsby-core-utils@3.1.2
 - gatsby-page-utils@2.1.2
 - gatsby-plugin-benchmark-reporting@2.1.2
 - gatsby-plugin-gatsby-cloud@4.1.2
 - gatsby-plugin-image@2.1.2
 - gatsby-plugin-manifest@4.1.3
 - gatsby-plugin-mdx@3.1.3
 - gatsby-plugin-offline@5.1.3
 - gatsby-plugin-page-creator@4.1.3
 - gatsby-plugin-preload-fonts@3.1.2
 - gatsby-plugin-sharp@4.1.3
 - gatsby-plugin-typescript@4.1.2
 - gatsby-recipes@1.1.2
 - gatsby-remark-images@6.1.3
 - gatsby-source-contentful@6.1.3
 - gatsby-source-drupal@5.1.3
 - gatsby-source-filesystem@4.1.2
 - gatsby-source-graphql@4.1.2
 - gatsby-source-shopify@6.1.3
 - gatsby-source-wordpress@6.1.2
 - gatsby-telemetry@3.1.2
 - gatsby-transformer-remark@5.1.3
 - gatsby-transformer-sqip@4.1.3
 - gatsby@4.1.4
 ```

**Pull request merged**